proto/mls/message_contents/group_metadata.proto

```diff
@@ -0,0 +1,63 @@
+// Group metadata
+syntax = "proto3";
+
+package xmtp.mls.message_contents;
+
+option go_package = "github.com/xmtp/proto/v3/go/mls/message_contents";
+option java_package = "org.xmtp.proto.mls.message.contents";
+
+// Parent message for group metadata
+message GroupMetadataV1 {
+  ConversationType conversation_type = 1;
+  string creator_account_address = 2;
+  PolicySet policies = 3;
+}
+
+// Defines the type of conversation
+enum ConversationType {
+  CONVERSATION_TYPE_UNSPECIFIED = 0;
+  CONVERSATION_TYPE_GROUP = 1;
+  CONVERSATION_TYPE_DM = 2;
+}
+
+// The set of policies that govern the group
+message PolicySet {
+  MembershipPolicy add_member_policy = 1;
+  MembershipPolicy remove_member_policy = 2;
+  MembershipPolicy add_installation_policy = 3;
+  MembershipPolicy remove_installation_policy = 4;
```

Comment:
Is there any reason the add/remove installation policies would be different between different conversations? Shouldn't this be considered baked into the protocol itself, rather than configurable?

Reply:
So, my thinking here is that we should give ourselves flexibility on the read side but not on the write side.

We'll have some prebaked combination of these policies that are the only options available to users (likely one to start). But if we want to roll out new prebaked options (for example, a chat where anyone can add members), we don't have to make a breaking change to the proto and wait for all clients to upgrade in order to roll it out. Only the group creator's app would need to be upgraded to the version with that new option available.

Reply:
Having the flexibility on the read side means that the write side can be modified by anyone (regardless of if we expose it via our SDK, they can set the payload themselves). Because installation adding/removing is tied in with our security and reliability guarantees, it feels unpredictable to me what can be done by tampering with these fields. Do you have any examples of how we might want to do adding/removing different for installations specifically in the future? I think the example you have is for adding/removing members which totally makes sense.

Reply:
OK. I see where you're coming from. Given that we don't _need_ configurability for installation add/remove, why allow for more possible diversity that would be hard to debug (even if attacks using this approach would be very unlikely).

I could be on board with making policies user-configurable only for add/remove members. We'd want to be super-sure there aren't any possibly valid cases where we'd want to change add/remove installations behaviour, but I can't think of any right now.